### PR TITLE
fix inbox loading not updating

### PR DESCRIPTION
**Pull request by chrisnojima**
The inbox loading bar can get stuck showing sometimes. This was cause the prop was correct but the list didn't know to re-render.

Instead of all the wackiness of making this work with the old NativeListView I updated this to use the FlatList. I also made the loading bar not push the list down a pixel while showing

### CODE REVIEW

File: shared/actions/chat/inbox.js [hunk 1]
```diff
@@ -32,7 +32,6 @@ let _backgroundLoopTask
 // Load the inbox if we haven't yet, mostly done by the UI
 function * onInitialInboxLoad (): SagaGenerator<any, any> {
   try {
-    yield put(Creators.setInboxUntrustedState('loading'))
```
Comment (chrisnojima) [inline shared/actions/chat/inbox.js]: We don't happen to need this as the inboxStale does this itself

File: shared/chat/inbox/index.native.js [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
+    style={{...globalStyles.flexBoxColumn, minHeight: 48, position: 'relative'}}>
     <ClickableBox style={{...globalStyles.flexBoxColumn, flex: 1, flexShrink: 0}} onClick={onNewChat}>
       <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', justifyContent: 'center', flex: 1}}>
         <Icon type='iconfont-new' style={{color: globalColors.blue, marginRight: 9}} />
         <Text type='BodyBigLink'>New chat</Text>
       </Box>
     </ClickableBox>
-    {isLoading && <LoadingLine />}
+    {isLoading && (
+      <Box style={{bottom: 0, left: 0, position: 'absolute', right: 0}}>
```
Comment (chrisnojima) [inline shared/chat/inbox/index.native.js]: don't push the list down when it shows

File: shared/chat/inbox/index.native.js [hunk 3]
```diff
@@ -209,12 +208,12 @@ class ConversationList extends PureComponent<void, Props, {dataSource: any}> {
   render () {
     return (
       <Box style={boxStyle}>
-        <NativeListView
-          enableEmptySections={true}
-          style={listStyle}
-          dataSource={this.state.dataSource}
-          onChangeVisibleRows={this._onChangeVisibleRows}
-          renderRow={this._itemRenderer} />
+        <FlatList
```
Comment (chrisnojima) [inline shared/chat/inbox/index.native.js]: FlatList ftw

File: shared/chat/inbox/index.native.js [hunk 4]
```diff
@@ -209,12 +208,12 @@ class ConversationList extends PureComponent<void, Props, {dataSource: any}> {
...
+          loading={this.props.isLoading/* force loading to update */}
```
Comment (chrisnojima) [inline shared/chat/inbox/index.native.js]: pass this down to force the list to re-render

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
- [x] Added fix for newly provisioned android not loading. On mount we try and unbox the first items. 
- [x] Added the keyExtractor else we get console.warns from FlatList

File: shared/chat/inbox/index.native.js [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
 import {debounce} from 'lodash'
+// $FlowIssue
+import FlatList from '../../fixme/Lists/FlatList'
 
 import type {Props, RowProps} from './'
 
 const AddNewRow = ({onNewChat, isLoading}: {onNewChat: () => void, isLoading: boolean}) => (
   <Box
-    style={{...globalStyles.flexBoxColumn, minHeight: 48}}>
+    style={{...globalStyles.flexBoxColumn, minHeight: 48, position: 'relative'}}>
```
Comment (gabriel) [inline shared/chat/inbox/index.native.js]: Isn't position relative default in RN?

File: shared/chat/inbox/index.native.js [hunk 6]
```diff
@@ -191,30 +195,35 @@ class ConversationList extends PureComponent<void, Props, {dataSource: any}> {
...
+    const item = viewableItems && viewableItems[0]
+    if (item && item.index) {
+      this._askForUnboxing(item.item, viewableItems.length)
     }
   }, 1000)
 
   componentWillMount () {
     this.props.loadInbox()
     this._setupDataSource(this.props)
+    if (this.props.rows.count()) {
```
Comment (gabriel) [inline shared/chat/inbox/index.native.js]: Maybe add `> 0`?

**Pull request merged**